tests/database/test_eth1_chaindb.py

```diff
@@ -291,3 +292,19 @@ def test_chaindb_persist_unexecuted_block(chain,
             NUMBER_BLOCKS_IN_CHAIN,
             TRANSACTIONS_IN_BLOCK + 1,
         )
+
+
+def test_chaindb_raises_blocknotfound_on_missing_uncles(VM, chaindb, header):
```

Comment:
Importantly, this tests against all VMs, since the implementation is currently sitting in the `Frontier` vm.